src/Raven.Client/Documents/Queries/Facets/RangeFacet.cs

```diff
@@ -269,17 +276,25 @@ private static string GetStringRepresentation(string fieldName, ExpressionType l
             if (lValue != null && rValue != null)
             {
                 if (leftOp == ExpressionType.GreaterThanOrEqual && rightOp == ExpressionType.LessThanOrEqual)
-                    return $"{fieldName} between {GetStringValue(lValue)} and {GetStringValue(rValue)}";
+                    return $"{fieldName} between {GetStringValue(lValue, parameters)} and {GetStringValue(rValue, parameters)}";
 
-                return $"{GetStringRepresentation(fieldName, leftOp, lValue)} and {GetStringRepresentation(fieldName, rightOp, rValue)}";
+                return $"{GetStringRepresentation(fieldName, leftOp, lValue, parameters)} and {GetStringRepresentation(fieldName, rightOp, rValue, parameters)}";
             }
 
             throw new InvalidOperationException("Unable to parse the given operation into a facet range!!! ");
         }
 
-        private static string GetStringValue(object value)
+        private static string GetStringValue(object value, Parameters parameters)
         {
-            switch (value.GetType().FullName)
+            var type = value.GetType().FullName;
+
+            if (parameters != null && 
+                type.In(new[] { "System.DateTime", "System.DateTimeOffset", "System.String" }))
```

Comment:
Avoid doing things this way, instead, use an `is DateTime`, `is DateTimeOffset`, etc.

Reply:
Why is this needed, anyway?